addons/l10n_uy/models/account_move.py

```diff
@@ -22,3 +32,19 @@ def _get_last_sequence_domain(self, relaxed=False):
             where_string += " AND l10n_latam_document_type_id = %(l10n_latam_document_type_id)s"
             param['l10n_latam_document_type_id'] = self.l10n_latam_document_type_id.id or 0
         return where_string, param
+
+    def _get_l10n_latam_documents_domain(self):
+        """ If this is a reversal or debit, suggest only related subtypes """
+        self.ensure_one()
+        domain = super()._get_l10n_latam_documents_domain()
+        if self.journal_id.country_code == "UY":
+            # We do this only to be able to match with the subtypes
+            if original_document := self.reversed_entry_id or self.debit_origin_id:
+                codes = self.env["l10n_latam.document.type"].search(domain).mapped('code')
+                for sublist_type in UY_DOC_SUBTYPES:
+                    if original_document.l10n_latam_document_type_id.code in sublist_type:
+                        break
```

Comment:
We can not do a filter instead?  (if it does not break, I think you will have the last one)